tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/strategy/ProgrammeBasedFilterStrategy.java

```diff
@@ -3,33 +3,80 @@
 import com.transformuk.hee.tis.security.model.Programme;
 import com.transformuk.hee.tis.security.model.UserProfile;
 import com.transformuk.hee.tis.security.util.TisSecurityHelper;
+import com.transformuk.hee.tis.tcs.api.enumeration.ProgrammeMembershipStatus;
+import com.transformuk.hee.tis.tcs.service.model.ColumnFilter;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
+import java.util.stream.Collectors;
 import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
 import org.apache.lucene.search.join.ScoreMode;
 import org.elasticsearch.common.collect.Tuple;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.index.query.MatchQueryBuilder;
 import org.elasticsearch.index.query.NestedQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
 import org.springframework.stereotype.Component;
 
 @Component
 public class ProgrammeBasedFilterStrategy implements RoleBasedFilterStrategy {
 
-  private static final String COLUMN_FILTER = "programmeName";
+  private static final String COLUMN_FILTER = "programmeId";
 
   @Override
-  public Optional<Tuple<String, BoolQueryBuilder>> getFilter() {
+  public Optional<Tuple<String, BoolQueryBuilder>> getFilter(List<ColumnFilter> columnFilters) {
+    MatchQueryBuilder statusMatchQueryBuilder = null;
+    if (columnFilters != null) {
+      Set<ColumnFilter> statusFilters = columnFilters.stream().filter(
+              columnFilter -> StringUtils.equals(columnFilter.getName(), "programmeMembershipStatus"))
+          .collect(
+              Collectors.toSet());
+      if (statusFilters.size() > 0) {
```

Comment:
If you wanted to get really funky, you could use a `Stream.map` to return an `Optional<MatchQueryBuilder> statusMatchQueryBuilder`.

Reply:
> N.B. As there is no handling for multiple filters of the same column, that would accept potentially let deterministic behaviour if there 2 filters of the same name were passed in.

I think this is a general problem for all APIs working with column filters and has to be handled in the same way.

Reply:
the N.B. should read

> ...potentially **less** deterministic...

i.e. the suggesstion to use a `Stream` might interpret the parameter differently across invocations because a Stream doesn't have the ordering constraints of a List.  I'm happy with that.